components/service-waiter/cmd/component.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+var componentCmdOpt struct {
+	image     string
+	namespace string
+	component string
+	labels    string
+}
+
+var componentCmd = &cobra.Command{
+	Use:   "component",
+	Short: "waits for component to become latest build of current installer build",
+	PreRun: func(cmd *cobra.Command, args []string) {
+		err := viper.BindPFlags(cmd.Flags())
+		if err != nil {
+			log.WithError(err).Fatal("cannot bind Viper to pflags")
+		}
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		if componentCmdOpt.image == "" {
+			log.Fatal("Target image should be defined")
```

Comment:
I wondered how to have this constraint in a way that it does not block us for hotfixes, but I could not come up with sth sensible.
IMO instead we should make it easy for operators to opt-out of the constraint here, in case we mitigate an incident or so.
@mustard-mh Do you think we could emit a warning in this case instead, so we can manually set the `image` to "" in such situations? It's complexity for the OP, but I see no way around it. :+1:

Reply:
@mustard-mh let's add such exception and document in the manual deployment?

Reply:
- Document changed [[notion](https://www.notion.so/gitpod/How-we-deploy-WebApp-9455c1f51ca74f29855c354c3f03c6c8?pvs=4#ce02850355194585b21abf68a3827480)]
- Skip service-waiter when image is empty https://github.com/gitpod-io/gitpod/pull/18995/commits/9d8e15a85ecf23be88a80023f1a6a7b23b74b3a1